crates/sui-core/src/checkpoints2/mod.rs

```diff
@@ -216,41 +251,220 @@ impl CheckpointBuilder {
     }
 }
 
+impl CheckpointAggregator {
+    fn new(
+        tables: Arc<CheckpointStoreTables>,
+        notify: Arc<Notify>,
+        exit: watch::Receiver<()>,
+        committee: Committee,
+        output: Box<dyn CertifiedCheckpointOutput>,
+    ) -> Self {
+        let current = None;
+        Self {
+            tables,
+            notify,
+            exit,
+            current,
+            committee,
+            output,
+        }
+    }
+
+    async fn run(mut self) {
+        loop {
+            if let Err(e) = self.run_inner().await {
+                error!(
+                    "Error while aggregating checkpoint, will retry in 1s: {:?}",
+                    e
+                );
+                tokio::time::sleep(Duration::from_secs(1)).await;
+                continue;
+            }
+
+            match select(self.exit.changed().boxed(), self.notify.notified().boxed()).await {
+                Either::Left(_) => {
+                    // return on exit signal
+                    return;
+                }
+                Either::Right(_) => {}
+            }
+        }
+    }
+
+    async fn run_inner(&mut self) -> SuiResult {
+        'outer: loop {
+            let current = if let Some(current) = &mut self.current {
+                current
+            } else {
+                let next_to_certify = self
+                    .tables
+                    .certified_checkpoints
+                    .iter()
+                    .skip_to_last()
+                    .next()
+                    .map(|(seq, _)| seq + 1)
+                    .unwrap_or_default();
+                let Some(summary) = self.tables.checkpoint_summary.get(&next_to_certify)? else { return Ok(()); };
```

Comment:
When is this table updated?

Reply:
`CheckpointBuilder` writes to this table

Reply:
Could you point me to the line? Couldn't find it.
Also, does that mean you would only start accepting signatures after you have signed yourself?

Reply:
`CheckpointBuilder::create_checkpoint` on line 197 writes there

>Also, does that mean you would only start accepting signatures after you have signed yourself?

Yes. However, once we get data sync it is also possible to just directly insert into `certified_checkpoints` and then this code will be completely skipped